tests/agent_features/test_configuration.py

```diff
@@ -583,3 +597,380 @@ def test_default_values(name, expected_value):
+function = function_to_trace
+"""
+
+wsgi_newrelic_ini = b"""
+[wsgi-application:]
+enabled = True
+function = test_configuration:function_to_trace
+application = app
+"""
+
+bad_wsgi_newrelic_ini = b"""
+[wsgi-application:]
+enabled = True
+function = function_to_trace
+application = app
+"""
+
+wsgi_missing_enabled_newrelic_ini = b"""
+[wsgi-application:]
+function = function_to_trace
+application = app
+"""
+
+wsgi_unparseable_enabled_newrelic_ini = b"""
+[wsgi-application:]
+enabled = not-a-bool
+function = function_to_trace
+application = app
+"""
+
+
+@pytest.mark.parametrize(
+    "section,expect_error",
+    (
+        (func_newrelic_ini, False),
+        (bad_func_newrelic_ini, True),
+        (func_missing_enabled_newrelic_ini, False),
+        (external_newrelic_ini, False),
+        (bad_external_newrelic_ini, True),
+        (external_missing_enabled_newrelic_ini, False),
+        (generator_newrelic_ini, False),
+        (bad_generator_newrelic_ini, True),
+        (generator_missing_enabled_newrelic_ini, False),
+        (bg_task_newrelic_ini, False),
+        (bad_bg_task_newrelic_ini, True),
+        (bg_task_missing_enabled_newrelic_ini, False),
+        (db_trace_newrelic_ini, False),
+        (bad_db_trace_newrelic_ini, True),
+        (db_trace_missing_enabled_newrelic_ini, False),
+        (wsgi_newrelic_ini, False),
+        (bad_wsgi_newrelic_ini, True),
+        (wsgi_missing_enabled_newrelic_ini, False),
+        (wsgi_unparseable_enabled_newrelic_ini, True),
+    ),
+    ids=(
+        "func_newrelic_ini",
+        "bad_func_newrelic_ini",
+        "func_missing_enabled_newrelic_ini",
+        "external_newrelic_ini",
+        "bad_external_newrelic_ini",
+        "external_missing_enabled_newrelic_ini",
+        "generator_newrelic_ini",
+        "bad_generator_newrelic_ini",
+        "generator_missing_enabled_newrelic_ini",
+        "bg_task_newrelic_ini",
+        "bad_bg_task_newrelic_ini",
+        "bg_task_missing_enabled_newrelic_ini",
+        "db_trace_newrelic_ini",
+        "bad_db_trace_newrelic_ini",
+        "db_trace_missing_enabled_newrelic_ini",
+        "wsgi_newrelic_ini",
+        "bad_wsgi_newrelic_ini",
+        "wsgi_missing_enabled_newrelic_ini",
+        "wsgi_unparseable_enabled_newrelic_ini",
+    ),
+)
+def test_initialize_developer_mode(section, expect_error, logger):
+    settings = global_settings()
+    apply_config_setting(settings, "monitor_mode", False)
+    apply_config_setting(settings, "developer_mode", True)
+    _reset_configuration_done()
+    _reset_instrumentation_done()
+    _reset_config_parser()
+
+    with tempfile.NamedTemporaryFile() as f:
+        f.write(newrelic_ini_contents)
+        f.write(section)
+        f.seek(0)
+
+        initialize(config_file=f.name)
+
+    if expect_error:
+        assert "CONFIGURATION ERROR" in logger.caplog.records
+    else:
+        assert "CONFIGURATION ERROR" not in logger.caplog.records
+
+
+@pytest.fixture
+def caplog_handler():
```

Comment:
I wonder if we should make this it's own fixture in testing support since we use it in the logging tests as well.